source/scripts/Background/controllers/MainController.ts

```diff
@@ -648,24 +647,37 @@ const MainController = (walletState): IMainController => {
 
     const currentAccount = accounts[activeAccount.type][activeAccount.id];
 
-    store.dispatch(setIsLoadingAssets(true));
-
     assetsManager.utils
       .updateAssetsFromCurrentAccount(
         currentAccount,
         isBitcoinBased,
         activeNetwork.url,
+        activeNetwork.chainId,
         networks
       )
       .then((updatedAssets) => {
+        const validateIfIsInvalidDispatch =
```

Comment:
this is really hard to read, maybe more constants with the validations and a final const with the final validation can make this cleaner